community/rfcs/techniques/design/20200306-weight-clustering-design.md

```diff
@@ -0,0 +1,144 @@
+# Keras Weight Clustering API Design
+
+| Status        | Accepted                                      |
+:---------------|:----------------------------------------------|
+| **Author(s)** | Mohamed Nour Abouelseoud (mohamednour.abouelseoud@arm.com), Aron Virginas-Tar (aron.virginas-tar@arm.com), Anton Kachatkou (anton.kachatkou@arm.com)|
+| **Sponsor**   | Alan Chiao (alanchiao@google.com)             |
+| **Updated**   | 2020-03-06                                    |
+
+See the initial proposal [here](../proposals/20200205-weight-clustering-proposal.md) for
+background and motivation for the technique.
+
+## Public API Summary
+
+The API is intended to be used in the same way the pruning API is used. The interface is intentionally made very similar to the one described in https://www.tensorflow.org/model_optimization/guide/pruning/.
+
+The clustering API will reveal the following new interfaces to the end user:
+
+* `cluster_weights()`
+
+  the main function to be used for clustering a Keras model or a layer. It configures either all supported layers in a model or a specific layer to be clustered;
+
+* `cluster_scope()`
+
+  provides a scope for deserializing clustered layers and models;
+
+* `strip_clustering()`
+
+  strips the clustering-specific information from a clustered model and restores the original model structure with the clustered weights.
+
+For further details about the API and how to use these functions, please refer to the Pydoc and the [examples below](#tutorials-and-examples).
+
+## Module Structure
```

Comment:
How the APIs are exposed to the public, which is the focus of the RFC, is more important the the file names. Note that for pruning, how the APIs are exposed is defined under python/core/api, whereas the files live under python/core/sparsity. 

So something like the sentence, "All APIs are exposed in the tfmot.sparsity.keras package (e.g. tfmot.sparsity.keras.prune_low_magnitude). The following APIs are available for different aspects of using pruning." from the [pruning RFC](https://github.com/tensorflow/model-optimization/pull/268/files?short_path=edfd107#diff-edfd10773bb6bb7662ce4fb0b94d12a1)

I'm assuming you would also propose to expose weight clustering as 
tfmot.clustering.keras.cluster_weights
tfmot.clustering.keras.strip_pruning
in the pip package.

Otherwise, make it clear. 

I'd remove the file names and paths (and the test files) since those are implementation details that are much more flexible than the public exposure itself.